src/error.jl

```diff
@@ -163,8 +163,23 @@ Assert if all the locations are:
 macro assert_locs_contiguous(n, locs, msgs...)
     msg = process_msgs(msgs...; default="locations is not contiguous.")
     quote
-        @assert_locs $(esc(n)) $(esc(locs))
+        @assert_locs_safe $(esc(n)) $(esc(locs))
         islocs_contiguous($(esc(n)), $(esc(locs))) || error($msg)
         nothing
     end
 end
+
+"""
+    @assert_locs_fit <number of qubits> <locations list> [<msg>]
+
+Assert if the location is exactly fit into n qubits.
+"""
+macro assert_locs_fit(n, locs, msgs...)
+    msg = process_msgs(msgs...; default="locations is not contiguous.")
+    quote
+        @assert_locs_safe $(esc(n)) $(esc(locs))
+        addrs |> first |> minimum == 1 || throw(AddressConflictError("addr not exact fit at 1"))
+        islocs_contiguous($(esc(n)), $(esc(locs))) || error($msg)
```

Comment:
addrs |> last |> maximum == n || throw(AddressConflictError("addr not exact fit at end"))

Reply:
I copied this from Yao v0.3.2 why change the order?

Reply:
you need to check both start and end.

Reply:
And, you need to sort first.

Reply:
And please write tests.